drivers/serial/uart_stm32.c

```diff
@@ -178,6 +178,36 @@ static inline u32_t uart_stm32_get_databits(struct device *dev)
 #endif	/* CONFIG_LPUART_1 */
 }
 
+static inline void uart_stm32_set_hwctrl(struct device *dev, u32_t hwctrl)
+{
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+
+#ifdef CONFIG_LPUART_1
+	if (IS_LPUART_INSTANCE(UartInstance)) {
+		LL_LPUART_SetHWFlowCtrl(UartInstance, hwctrl);
```

Comment:
It appears LL LPUART and LL USART APIs are quite close and identical in the way we use them.
So we can clean up part of this and call LL_USART_SetHWFlowCtrl for all (LP)U(S)ART instances.
Btw, this is already the case in various tx/rc related functions.

Additional clean up could also be done in recent configure functions, if you're in for some code optimization.
If needed we can put a comment somewhere to state that and warn that this is still the case with new LL API calls that may be introduced.

Reply:
I think such clean-up (touching all LPUART calls) would be better as a separate PR.
Reworked HW flow control related part, and added comments there for the future.